R/nfidd-stan-tools.r

```diff
@@ -0,0 +1,304 @@
+#' write_to_file is TRUE. Defaults to "nfidd_functions.stan".
+#'
+#' @return A character string containing the requested Stan functions
+#'
+#' @family stantools
+#'
+#' @export
+nfidd_load_stan_functions <- function(
+    functions = NULL, stan_path = nfidd::nfidd_stan_path(),
+    wrap_in_block = FALSE, write_to_file = FALSE,
+    output_file = "nfidd_functions.stan") {
+  stan_files <- list.files(
+    file.path(stan_path, "functions"),
+    pattern = "\\.stan$", full.names = TRUE,
+    recursive = TRUE
+  )
+  all_content <- character(0)
+
+  for (file in stan_files) {
+    content <- readLines(file)
+    if (is.null(functions)) {
+      all_content <- c(all_content, content)
+    } else {
+      for (func in functions) {
+        func_content <- .extract_stan_functions(
+          content,
+          names_only = FALSE,
+          functions = func
+        )
+        all_content <- c(all_content, func_content)
+      }
+    }
+  }
+
+  # Add version comment
+  version_comment <- paste(
+    "// Stan functions from nfidd version",
+    utils::packageVersion("nfidd")
+  )
+  all_content <- c(version_comment, all_content)
+
+  if (wrap_in_block) {
+    all_content <- c("functions {", all_content, "}")
+  }
+
+  result <- paste(all_content, collapse = "\n")
+
+  if (write_to_file) {
+    writeLines(result, output_file)
+    message("Stan functions written to: ", output_file, "\n")
+  }
+
+  return(result)
+}
+
+#' List Available Stan Models in NFIDD
+#'
+#' This function finds all available Stan models in the NFIDD package and
+#' returns their names without the .stan extension.
+#'
+#' @param stan_path Character string specifying the path to Stan files. Defaults
+#'   to the result of `nfidd_stan_path()`.
+#'
+#' @return A character vector of available Stan model names.
+#'
+#' @export
+#'
+#' @examples
+#' nfidd_list_stan_models()
+nfidd_stan_models <- function(
+    stan_path = nfidd::nfidd_stan_path()
+  ) {
+  stan_files <- list.files(
+    stan_path,
+    pattern = "\\.stan$", full.names = FALSE,
+    recursive = FALSE
+  )
+
+  # Remove .stan extension
+  model_names <- tools::file_path_sans_ext(stan_files)
+
+  return(model_names)
+}
+
+#' Create a CmdStanModel with NFIDD Stan functions
+#'
+#' This function creates a CmdStanModel object using a specified Stan model from
+#' the NFIDD package and optionally includes additional user-specified Stan
+#' files.
+#'
+#' @param model_name Character string specifying which Stan model to use.
+#' @param include_paths Character vector of paths to include for Stan
+#'  compilation. Defaults to the result of `nfidd_stan_path()`.
+#' @param ... Additional arguments passed to cmdstanr::cmdstan_model().
+#'
+#' @return A CmdStanModel object.
+#'
+#' @export
+#'
+#' @family modelhelpers
+#'
+#' @examplesIf requireNamespace("cmdstanr", quietly = TRUE)
+#' if (!is.null(cmdstanr::cmdstan_version(error_on_NA = FALSE))) {
+#'   model <- nfidd_cmdstan_model("renewal", compile = FALSE)
```

Comment:
```suggestion
#'   model <- nfidd_cmdstan_model("simple-nowcast", compile = FALSE)
```
or any other one that actually exists?